wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsWorkUnitsClient.java

```diff
@@ -700,9 +673,8 @@ public WUQueryResponse workUnitUQuery(String wuid, String jobname, String cluste
             if (jobname != null) request.setJobname(jobname);
             if (count != null) request.setCount(count);
             if (owner != null) request.setOwner(owner);
-            if (appKey != null) request.setApplicationKey(appKey);
-            if (appData != null) request.setApplicationData(appData);
-            if (appName != null) request.setApplicationName(appName);
+            ApplicationValue applicationValue[] = {new ApplicationValue(appKey,appName,appData)};
```

Comment:
Not at the moment.  I was thinking of adding a signature in that would replace the appName, appKey, appData parameters with an ApplicationValue[] parameter.  I'd use the code from the current implementation and then just make the signature we have currently (for backwards compatibility) be a wrapper to the new sig.